src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkUnrecognizedSectionInspection.kt

```diff
@@ -46,17 +56,29 @@ class SmkUnrecognizedSectionInspection : SnakemakeInspection() {
          */
         private fun isSectionRecognized(
             argsSection: SmkArgsSection,
-            setOfValidNames: Set<String>
+            setOfValidNames: Set<String>,
+            sectionName: String
         ) {
             val sectionNamePsi = argsSection.nameIdentifier
             val sectionKeyword = argsSection.sectionKeyword
 
             if (sectionNamePsi != null && sectionKeyword != null && sectionKeyword !in setOfValidNames
                 && sectionKeyword !in ignoredItems
             ) {
+                val appropriateSection = getSectionBySubsection(sectionKeyword)
+                val message = if (appropriateSection == null) {
+                    SnakemakeBundle.message("INSP.NAME.section.unrecognized.message", sectionKeyword)
+                } else {
+                    SnakemakeBundle.message(
+                        "INSP.NAME.section.unexpected",
+                        sectionKeyword,
+                        sectionName,
+                        appropriateSection
+                    )
+                }
                 registerProblem(
                     sectionNamePsi,
-                    SnakemakeBundle.message("INSP.NAME.section.unrecognized.message", sectionKeyword),
+                    message,
```

Comment:
This implementation hardcodes problem type and ignores IDE settings, let's use `ProblemHighlightType.GENERIC_ERROR_OR_WARNING` instead.